### PR TITLE
Post block v2 integration tests

### LINKED ISSUE #7349: Add new block publishing API with `broadcast_validation` param
https://github.com/ethereum/beacon-APIs/pull/317

Note: we need to take care of respecting `202` http status code condition.

### PR DESCRIPTION
**Pull request by tbenr**
This is the last PR.

Adds integration tests.

Fixes https://github.com/Consensys/teku/issues/7349

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/BeaconRestApiTypesTest.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import org.junit.jupiter.params.provider.EnumSource;
+import tech.pegasys.teku.beaconrestapi.BeaconRestApiTypes.BroadcastValidationParameter;
+
+public class BeaconRestApiTypesTest {
+
+  @ParameterizedTest
+  @EnumSource(BroadcastValidationParameter.class)
+  void broadcastValidationParameter_anInternalVersionMustExist(
+      final BroadcastValidationParameter parameter) {
+    parameter.toInternal();
```
Comment (zilm13) [inline data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/BeaconRestApiTypesTest.java]: A bit confusing, what is it for? Maybe wrap in `assertNotThrow(() ->`?

File: data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/validator/PostBlindedAndUnblindedBlockTest.java [hunk 2]
```diff
@@ -141,14 +183,18 @@ void shouldReturnOkPostDeneb(
...
+              milestone)) {
         assertThat(response.code()).isEqualTo(SC_OK);
       }
     } else {
       try (final Response response =
           post(
               route,
-              JsonUtil.serialize(request, signedBlockContainerSchema.getJsonTypeDefinition()))) {
+              JsonUtil.serialize(request, signedBlockContainerSchema.getJsonTypeDefinition()),
+              params)) {
```
Comment (zilm13) [inline data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/validator/PostBlindedAndUnblindedBlockTest.java]: you could add milestone in header to json also for uniformity

**Review by zilm13 — APPROVED** [review]
LGTM, just few nits

File: data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/BeaconRestApiTypesTest.java (hunk 1 above)
Comment (tbenr) [reply data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/BeaconRestApiTypesTest.java]: ah yes, would me more clear

**Pull request merged**